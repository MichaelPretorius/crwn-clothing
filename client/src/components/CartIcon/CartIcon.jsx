import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { CartIconContainer, ItemCountContainer, ShoppingIconContainer } from './CartIcon.styles';
import { toggleCartHidden } from '../../redux/cart/cartActions';
import { selectCartItemsCount } from '../../redux/cart/cartSelectors';

const CartIcon = ({ toggleCartHidden, itemCount }) => {
  return (
    <CartIconContainer onClick={toggleCartHidden}>
      <ShoppingIconContainer className="shopping-icon" />
      <ItemCountContainer>{itemCount}</ItemCountContainer>
    </CartIconContainer>
  );
}

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

export default connect(mapStateToProps, { toggleCartHidden })(CartIcon);