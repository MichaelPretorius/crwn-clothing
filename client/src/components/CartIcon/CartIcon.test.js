import React from 'react';
import { shallow } from 'enzyme';
import { CartIcon } from './CartIcon';

describe('CartIcon', () => {
  let wrapper;
  let mockToggleCartHidden;
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  test('should render CartIcon component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  
  test('should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartIconContainer').simulate('click');
    expect(mockToggleCartHidden).toBeCalled();
  });
  
  test('should render itemCount as the text', () => {
    const itemCount = parseInt(wrapper.find('ItemCountContainer').text());
    expect(itemCount).toEqual(0);
  });
})
