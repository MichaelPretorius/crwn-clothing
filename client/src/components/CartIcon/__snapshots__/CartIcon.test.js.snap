// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon should render CartIcon component 1`] = `
"<CartIconContainer onClick={[Function: mockConstructor]}>
  <Styled(Component) className=\\"shopping-icon\\" />
  <ItemCountContainer>
    0
  </ItemCountContainer>
</CartIconContainer>"
`;
